core/domain/blog_validators.py

```diff
@@ -53,20 +67,59 @@ def _get_external_id_relationships(cls, item):
         ]
         field_name_to_external_model_references.append(
             base_model_validators.UserSettingsModelFetcherDetails(
-                'author_ids', [item.author_id],
+                'author_id', [item.author_id],
                 may_contain_system_ids=True,
                 may_contain_pseudonymous_ids=True
             )
         )
         return field_name_to_external_model_references
 
+    @classmethod
+    def _validate_title_is_unique(cls, item):
+        """Validate that title of the model unique.
+
+        Args:
+            item: datastore_services.Model. BlogPostModel to validate.
+        """
+        if item.title != '':
+            blog_post_models = blog_models.BlogPostModel.query().filter(
+                blog_models.BlogPostModel.title == (
+                    item.title)).filter(
+                        blog_models.BlogPostModel.deleted == False).fetch() # pylint: disable=singleton-comparison
+            blog_model_ids = [
+                model.id for model in blog_post_models if model.id != item.id]
+            if blog_model_ids:
+                cls._add_error(
+                    'unique title for blog post',
+                    'Entity id %s: title %s matches with title '
+                    'blog post models with ids %s' % (
+                        item.id, item.title, blog_model_ids))
+
+    @classmethod
+    def _get_custom_validation_functions(cls):
+        return [cls._validate_title_is_unique]
```

Comment:
Are we expecting to validate the URL fragment is unique?

Reply:
Added